lib/assert.js

```diff
@@ -369,9 +381,157 @@ function isSpecial(obj) {
 }
 
 const typesToCallDeepStrictEqualWith = [
-  isKeyObject, isWeakSet, isWeakMap, Buffer.isBuffer,
+  isKeyObject, isWeakSet, isWeakMap, Buffer.isBuffer, isSharedArrayBuffer,
 ];
 
+function compareMaps(actual, expected, comparedObjects) {
+  if (actual.size !== expected.size) {
+    return false;
+  }
+  const safeIterator = FunctionPrototypeCall(SafeMap.prototype[SymbolIterator], actual);
+
+  comparedObjects ??= new SafeWeakSet();
+
+  for (const { 0: key, 1: val } of safeIterator) {
+    if (!MapPrototypeHas(expected, key)) {
+      return false;
+    }
+    if (!compareBranch(val, MapPrototypeGet(expected, key), comparedObjects)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function compareArrayBuffers(actual, expected) {
+  let actualView, expectedView, expectedViewLength;
+
+  if (!ArrayBufferIsView(actual)) {
+    let actualViewLength;
+
+    if (isArrayBuffer(actual) && isArrayBuffer(expected)) {
+      actualViewLength = ArrayBufferPrototypeGetByteLength(actual);
+      expectedViewLength = ArrayBufferPrototypeGetByteLength(expected);
+    } else if (isSharedArrayBuffer(actual) && isSharedArrayBuffer(expected)) {
+      actualViewLength = actual.byteLength;
+      expectedViewLength = expected.byteLength;
+    } else {
+      // Cannot compare ArrayBuffers with SharedArrayBuffers
+      return false;
+    }
```

Comment:
We don't need to compute the value twice
```suggestion
    const isActualSharedArrayBuffer = isSharedArrayBuffer(actual);

    if (isActualSharedArrayBuffer !== isSharedArrayBuffer(expected)) return false;
    
    if (isActualSharedArrayBuffer) {
      actualViewLength = actual.byteLength;
      expectedViewLength = expected.byteLength;
    } else {
      // If it is a BufferView and not a SharedArrayBuffer, it has to be an ArrayBuffer.
      actualViewLength = ArrayBufferPrototypeGetByteLength(actual);
      expectedViewLength = ArrayBufferPrototypeGetByteLength(expected);
    }
```

Reply:
Hum actually we do want to validate that `expected` is indeed a `BufferView`
```suggestion
    const isActualSharedArrayBuffer = isSharedArrayBuffer(actual);
    if (!isActualSharedArrayBuffer && isArrayBuffer(expected)) {
      actualViewLength = ArrayBufferPrototypeGetByteLength(actual);
      expectedViewLength = ArrayBufferPrototypeGetByteLength(expected);
    } else if (isActualSharedArrayBuffer && isSharedArrayBuffer(expected)) {
      actualViewLength = actual.byteLength;
      expectedViewLength = expected.byteLength;
    } else {
      // Cannot compare ArrayBuffers with SharedArrayBuffers
      return false;
    }
```